### DISCUSSION TITLE
Warn before closing without saving

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1975] [Q&A]
There's a certain workflow that I'm doing with Excel that I want to do with VisiData instead. Every day I open a CSV file, edit some cells, and close it. 

One basic feature in a program like Excel is that if I attempt to close the program without saving, it prompts me to save. VisiData doesn't do that, which makes sense for most VisiData usage. However, for this specific usage, I would like VisiData to prompt me if I attempt to close it without saving. I know of the `guard-sheet` feature, but if I understand correctly it doesn't really look at whether the file was saved or not.

Is there a way to do that with VisiData?

**Comment by anjakefala** [comment]
Saving does flip the `hasBeenModified` attribute, which is what `options.quitguard` uses to determine whether to guard the sheet! I think setting `options.quitguard = True` should get you what you want.

**Reply by cool-RR** [reply]
I tried that, and it worked well, but then I attempted to turn in on using the `--quitguard` command line argument (because I only want it to happen on this specific workflow, not all of my files), and when I use the command line argument, the behavior is different. I get the `quit guarded sheet "Log"?` message even when the file is unchanged.

**Reply by anjakefala** [reply]
Sounds like a bug! Can you write a report?

**Reply by cool-RR** [reply]
Yep: https://github.com/saulpw/visidata/issues/1976
